### PR TITLE
Enable big query config in aks service

### PR DESCRIPTION
**Pull request by vipin-dfe**
### Trello card
https://trello.com/c/5YFXDRps

### Context
All services will be migrated to the workload identity federation for improved security.

### Changes proposed in this pull request
Enable the analytics module for get into teaching app

### Guidance to review
Events should be published for review apps into big query

### CODE REVIEW

**Comment by vipin-dfe** [comment]
[GIT: migrate to GCP WIF](https://trello.com/c/5YFXDRps/2117-git-migrate-to-gcp-wif)

File: config/initializers/dfe_analytics.rb [hunk 1]
```diff
@@ -61,4 +61,8 @@
     Rails.application.config.action_controller.perform_caching &&
       ActionDispatch::FileHandler.new(Rails.root.join("public/cached_pages").to_s).attempt(rack_env).present?
   end
+
+  config.bigquery_maintenance_window = "08-09-2024 18:00..08-09-2024 19:00"
```
Comment (saliceti) [inline config/initializers/dfe_analytics.rb:65]: Is that required?

File: terraform/aks/application.tf [hunk 2]
```diff
@@ -16,7 +16,10 @@ module "application_configuration" {
   is_rails_application = true
 
   config_variables = {
-    PGSSLMODE        = local.postgres_ssl_mode
+    PGSSLMODE           = local.postgres_ssl_mode
+    BIGQUERY_DATASET    = var.enable_dfe_analytics_federated_auth ? module.dfe_analytics[0].bigquery_dataset : null
```
Comment (saliceti) [inline terraform/aks/application.tf:20]: This doesn't matter if federated auth is enabled or not, as BIGQUERY_DATASET, BIGQUERY_PROJECT_ID and BIGQUERY_TABLE_NAME won't change

File: terraform/aks/dfe_analytics.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  namespace             = var.namespace
+  service_short         = var.service_short
+  environment           = local.environment
+  gcp_dataset           = var.dataset_name
+  # Required when creating a new dataset
+  gcp_keyring           = var.dataset_name == null ? "git-key-ring" : null
+  gcp_key               = var.dataset_name == null ? "git-key" : null
+  gcp_taxonomy_id       = var.dataset_name == null ? 6582996680844371982 : null
+  gcp_policy_tag_id     = var.dataset_name == null ? 2845706316283381915 : null
+}
```
Comment (saliceti) [inline terraform/aks/dfe_analytics.tf:20]: Missing end of file line break. This can easily be set up in the IDE

File: terraform/aks/variables.tf [hunk 4]
```diff
@@ -40,6 +43,14 @@ variable "enable_postgres_backup_storage" {
 variable "docker_image" {
   description = "Docker image full name to identify it in the registry. Includes docker registry, repository and tag e.g.: ghcr.io/dfe-digital/teacher-pay-calculator:673f6309fd0c907014f44d6732496ecd92a2bcd0"
 }
+variable "enable_dfe_analytics_federated_auth" {
+  description = "Create the resources in Google cloud for federated authentication and enable in application"
+  default     = false
+}
+variable "dataset_name" {
```
Comment (saliceti) [inline terraform/aks/variables.tf:50]: This must be set for each environment

File: terraform/aks/variables.tf [hunk 5]
```diff
@@ -19,6 +19,9 @@ variable "azure_resource_prefix" {
 variable "config_short" {
   description = "Short name of the environment configuration, e.g. dv, st, pd..."
 }
+variable "config" {
```
Comment (saliceti) [inline terraform/aks/variables.tf:22]: Is this required?

File: config/initializers/dfe_analytics.rb (hunk 1 above)
Comment (vipin-dfe) [reply config/initializers/dfe_analytics.rb]: Removed.

File: terraform/aks/application.tf (hunk 2 above)
Comment (vipin-dfe) [reply terraform/aks/application.tf]: Updated

File: terraform/aks/dfe_analytics.tf (hunk 3 above)
Comment (vipin-dfe) [reply terraform/aks/dfe_analytics.tf]: Added.

File: .env.rolling [hunk 6]
```diff
@@ -6,8 +6,5 @@ VWO_ID=713156
 SKYLIGHT_ENV=dev
 FAST_IMAGE_TIMEOUT=0.25
 SIZE_IMAGES=1
-BIGQUERY_TABLE_NAME=events
```
Comment (saliceti) [inline .env.rolling:9]: Shouldn't they be removed from the other .env files?

Comment (vipin-dfe) [reply .env.rolling:9]: Done.

File: terraform/aks/variables.tf [hunk 7]
```diff
@@ -40,6 +40,14 @@ variable "enable_postgres_backup_storage" {
 variable "docker_image" {
   description = "Docker image full name to identify it in the registry. Includes docker registry, repository and tag e.g.: ghcr.io/dfe-digital/teacher-pay-calculator:673f6309fd0c907014f44d6732496ecd92a2bcd0"
 }
+variable "enable_dfe_analytics_federated_auth" {
+  description = "Create the resources in Google cloud for federated authentication and enable in application"
+  default     = false
+}
+variable "dataset_name" {
+  description = "dfe analytics dataset name in Google Bigquery. If not specified, a new one will be created"
```
Comment (saliceti) [inline terraform/aks/variables.tf:48]: > If not specified, a new one will be created
No we won't create a new one

**Review by saliceti — APPROVED** [review]

**Pull request merged**